File: android/app/src/main/java/com/llmhub/llmhub/mimobot/speech/Tts.kt
package com.llmhub.llmhub.mimobot.speech

import kotlinx.coroutines.flow.Flow

/**
 * Abstraction over any text-to-speech backend. Implementations emit 16 kHz
 * mono 16-bit PCM in 320-sample frames so the caller can feed them straight
 * into [OpusEncoder].
 *
 * v0 ships two implementations:
 *   - [SystemTts]  — Android's built-in TextToSpeech (free, decent, ships on every phone)
 *   - [KokoroTts]  — Kokoro-82M ONNX (higher quality, larger download)
 *
 * Later additions (Piper, Moshi / Mini-Omni full speech-to-speech) implement
 * this interface too. The pipeline doesn't care which one is plugged in.
 */
interface Tts {
    suspend fun init()
    fun speakToPcm(text: String, language: String = "en-US"): Flow<ShortArray>
    fun stop()
    fun close()
}

File: ios/LLMHub/Sources/LLMHub/MimoBot/TTS.swift
import Foundation

/// Abstraction over any text-to-speech backend. Implementations emit 16 kHz
/// mono Int16 PCM in 320-sample frames ready for Opus encode.
///
/// v0 ships two implementations:
///   - `SystemTTS` — AVSpeechSynthesizer (free, decent, ships with iOS)
///   - `KokoroTTS` — Kokoro-82M via ONNX Runtime Core ML EP (higher quality)
protocol TTS: AnyObject {
    func load() async throws
    func speakToPCM(text: String, language: String) -> AsyncStream<[Int16]>
    func stop()
}
